### ISSUE TITLE
Cryptic thrift error message when using unknown endpoint name.

### ISSUE THREAD

**Issue by Raynos** [uber/tcurl#15]
```
tcurl -p localhost:51015 populous health -t ~/Uber/populous/thrift/

/Users/jho/.nvm/v0.10.26/lib/node_modules/tcurl/node_modules/tchannel/node_modules/bufrw/result.js:32
        throw self.err;
                  ^
Error: type health_args not found
```

We should warn users that `health` is an invalid endpoint because it doesnt have a service. Try `MyService::health`

cc @kriskowal @ShanniLi

**Comment by uber-liang** [comment]
+1 on fixing this issue. I'm a beginner of tchannel and I wasted 1+ hour on this error today.

**Comment by ShanniLi** [comment]
cc @leizha, that is not really a tcurl issue. thriftify should provide better error messages.

**Comment by Raynos** [comment]
@malandrew will look into fixing this;

This is a tcurl issue not a thriftify issue. It's all about tcurl user experience.

**Comment by malandrew** [comment]
Starting on this now.

**Comment by malandrew** [comment]
This issue appears to be resolved. Here's what I get when I try to reproduce it.

known good service and endpoint:
```
~/dsg/tchannel/node$ tcurl -p 127.0.0.1:21300 autobahn health_v1
'Got call response ok'
'hello from autobahn\n'
```

endpoint that doesn't exist, but service does:
```
~/dsg/tchannel/node$ tcurl -p 127.0.0.1:21300 autobahn health_v2
{ [TchannelBadRequestError: no such endpoint service="autobahn" endpoint="health_v2"]
  type: 'tchannel.bad-request',
  message: 'no such endpoint service="autobahn" endpoint="health_v2"',
  isErrorFrame: true,
  codeName: 'BadRequest',
  errorCode: 6,
  originalId: 2,
  name: 'TchannelBadRequestError',
  fullType: 'tchannel.bad-request' }
no such endpoint service="autobahn" endpoint="health_v2"
```

service doesn't exist:
```
~/dsg/tchannel/node$ tcurl -p 127.0.0.1:21300 autobahns health_v1
{ [TchannelDeclinedError: no peer available for request]
  type: 'tchannel.declined',
  message: 'no peer available for request',
  isErrorFrame: true,
  codeName: 'Declined',
  errorCode: 4,
  originalId: 2,
  name: 'TchannelDeclinedError',
  fullType: 'tchannel.declined' }
no peer available for request
```

**Comment by Raynos** [comment]
@malandrew this is a thrift only thing.

You need to go into thrift mode with `-t thriftFile` 

The JSON code has sensible error messages; this is only a problem for thrift endpoints.

**Comment by malandrew** [comment]
Closed by #33
